### PR TITLE
Adds Just build cache

### PR DESCRIPTION
**Pull request by kenotron**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This change adds build cache at the package level. It respects monorepo deps. It works even if rush cache fails - and is a necessary step to get us onto lerna / yarn workspaces to preserve this nice dev experience piece.

**Caveat**: `rush rebuild` will no longer work directly out of the box after this. This is a bit of a take back; however, we're moving onto another system for monorepo. We will respect `npm run build` being a rebuild and `npm run buildfast` to be build with cache (this is the same behavior as today, albeit kind of incorrectly named)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9794)

### CODE REVIEW

**Comment by kenotron** [comment]
@msft-github-bot please merge this in 1 minute

**Comment by kenotron** [comment]
removed the auto merge since the perf-test seems to be screwed up

**Comment by JasonGore** [comment]
I can't build this branch on Windows, even with a full clean.

image

File: change/@uifabric-styling-2019-07-11-15-35-18.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "comment": "Adding Georgian and Armenian font for Segoe UI Web",
```
Comment (JasonGore) [inline change/@uifabric-styling-2019-07-11-15-35-18.json]: Should this be in the PR?

File: change/office-ui-fabric-react-2019-07-11-15-35-18.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "comment": "Panel: remove window.innerHeight from styles",
```
Comment (JasonGore) [inline change/office-ui-fabric-react-2019-07-11-15-35-18.json]: And this

File: change/@uifabric-react-cards-2019-07-11-15-35-18.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "comment": "Card: Importing from office-ui-fabric-react directly instead of from office-ui-fabric-react/lib/...",
```
Comment (JasonGore) [inline change/@uifabric-react-cards-2019-07-11-15-35-18.json]: ?

**Comment by JasonGore** [comment]
The Windows error seems to be a bit of a red herring.. seems like there could be a missing `node` command somewhere.

https://stackoverflow.com/questions/37287666/gulp-run-fails-on-windows-error-message-microsoft-jscript-compilation-error

File: change/@uifabric-react-cards-2019-07-11-15-35-18.json (hunk 3 above)
Comment (kenotron) [reply change/@uifabric-react-cards-2019-07-11-15-35-18.json]: Just a precommit hook that auto converts these stray change files from rush format to beachball.

File: change/office-ui-fabric-react-2019-07-11-15-35-18.json (hunk 2 above)
Comment (kenotron) [reply change/office-ui-fabric-react-2019-07-11-15-35-18.json]: yeah those are from other PRs

Comment (kenotron) [reply change/office-ui-fabric-react-2019-07-11-15-35-18.json]: from auto-commit

**Comment by JasonGore** [comment]
Per our chat `rush rebuild` doesn't seem to do a full rebuild anymore:

```
SUCCESS (22)
================================
@uifabric/build (2.15 seconds)
@uifabric/example-app-base (2.46 seconds)
@uifabric/experiments (2.38 seconds)
@uifabric/prettier-rules (2.14 seconds)
@uifabric/set-version (2.51 seconds)
@uifabric/tslint-rules (1.88 seconds)
office-ui-fabric-react (2.59 seconds)
@uifabric/codepen-loader (2.30 seconds)
@uifabric/fluent-theme (2.38 seconds)
@uifabric/merge-styles (2.37 seconds)
@uifabric/styling (2.33 seconds)
@uifabric/variants (2.55 seconds)
@uifabric/utilities (2.40 seconds)
@uifabric/jest-serializer-merge-styles (2.50 seconds)
@uifabric/test-utilities (2.48 seconds)
@uifabric/foundation (2.71 seconds)
@uifabric/icons (2.64 seconds)
@uifabric/webpack-utils (2.51 seconds)
@uifabric/azure-themes (2.62 seconds)
@uifabric/file-type-icons (2.65 seconds)
@uifabric/theme-samples (2.38 seconds)
perf-test (14.74 seconds)
================================
```

**Comment by kenotron** [comment]
> Per our chat `rush rebuild` doesn't seem to do a full rebuild anymore:
> 
> ```
> SUCCESS (22)
> ================================
> @uifabric/build (2.15 seconds)
> @uifabric/example-app-base (2.46 seconds)
> @uifabric/experiments (2.38 seconds)
> @uifabric/prettier-rules (2.14 seconds)
> @uifabric/set-version (2.51 seconds)
> @uifabric/tslint-rules (1.88 seconds)
> office-ui-fabric-react (2.59 seconds)
> @uifabric/codepen-loader (2.30 seconds)
> @uifabric/fluent-theme (2.38 seconds)
> @uifabric/merge-styles (2.37 seconds)
> @uifabric/styling (2.33 seconds)
> @uifabric/variants (2.55 seconds)
> @uifabric/utilities (2.40 seconds)
> @uifabric/jest-serializer-merge-styles (2.50 seconds)
> @uifabric/test-utilities (2.48 seconds)
> @uifabric/foundation (2.71 seconds)
> @uifabric/icons (2.64 seconds)
> @uifabric/webpack-utils (2.51 seconds)
> @uifabric/azure-themes (2.62 seconds)
> @uifabric/file-type-icons (2.65 seconds)
> @uifabric/theme-samples (2.38 seconds)
> perf-test (14.74 seconds)
> ================================
> ```

Added a caveat in the description. Will send mail about this.

File: scripts/invalidate-just-cache.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const fs = require('fs-extra');
+const path = require('path');
+
+const topLevelDep = path.join(__dirname, '../packages/prettier-rules/node_modules/.just');
```
Comment (jdhuntington) [inline scripts/invalidate-just-cache.js]: Why does this pull from `prettier-rules` ?

File: scripts/package.json [hunk 5]
```diff
@@ -7,10 +7,10 @@
     "just-scripts": "./just-scripts.js"
   },
   "scripts": {
-    "build": "node -v",
+    "build": "node ./just-scripts.js no-op",
     "bundlesize": "bundlesize --debug",
     "clean": "",
-    "code-style": "npx just-scripts code-style",
+    "code-style": "node ./just-scripts.js code-style",
```
Comment (jdhuntington) [inline scripts/package.json]: :+1:

File: scripts/invalidate-just-cache.js (hunk 4 above)
Comment (kenotron) [reply scripts/invalidate-just-cache.js]: This file will go away in my next PR - but it just so happens to be the one package that is at the top level (so invalidating this cache invalidates all build caches for all packages)

**Pull request merged**